starport/services/chain/serve.go

```diff
@@ -299,6 +303,28 @@ func (c *Chain) serve(ctx context.Context, forceReset bool) error {
 		return err
 	}
 
+	// we also consider the binary in the checksum to ensure the binary has not been changed by a third party
+	binaryModified := false
+	binaryName, err := c.Binary()
+	if err != nil {
+		return err
+	}
+	binaryPath, err := exec.LookPath(binaryName)
+	if err != nil {
+		if strings.Contains(err.Error(), "file not found") {
+			binaryModified = true
+		} else {
+			return err
```

Comment:
Can we refactor the whole section by using early returns? I think code becomes much more readable that way.

Reply:
What do you mean with the "whole section"?
This part can actually be reversed for better clarity

Reply:
I mean something like this:

```go
binaryName, err := c.Binary()
if err != nil {
    return err
}

binaryPath, err := exec.LookPath(binaryName)

if !errors.Is(err, exec.ErrNotFound) {
	return err
}

binaryModified := err != nil

if err == nil {
	binaryModified, err = dirchange.HasDirChecksumChanged("", []string{binaryPath}, saveDir, binaryChecksum)
	if err != nil {
		return err
	}
}
```

Reply:
I personally find it a bit less clear. We check the err even if it's nil and `binaryModified := err != nil` is a bit ambiguous IMO

Reply:
What about the moving `errors.Is` to upper level part?

Reply:
What do you think of:
```go
binaryPath, err := exec.LookPath(binaryName)
if err != nil && !errors.Is(err, exec.ErrNotFound) {
       return err
}
if errors.Is(err, exec.ErrNotFound) {
       binaryModified = true
}  else {
       binaryModified, err = dirchange.HasDirChecksumChanged("", []string{binaryPath}, saveDir, binaryChecksum)
       if err != nil {
	      return err
       }
}
```

Reply:
I think current way is just fine. 👍